doc/manual/manual.tex

```diff
@@ -9681,6 +9681,69 @@ \subsubsection{Thin shell gravity benchmark}
 The accuracy of the calculations increases with $I$ but so does the time spent in the postprocessor: for $I\in\{-1,0,1,2\}$  this time was about 18, 132, 440 and 1040~\si{s} respectively. This is easily explained when one realizes that increasing $I$ from 0 to 1 means that the number of GLQ points per cell goes from $2^3=8$ to $3^3=27$, i.e. a $3.375$ increase in operations for the same number of cells and measurement points. The time spent in the postprocessor increases by a similar factor ($440/132\sim 3.4$).
 The accuracy obtained with lower $I$ values increases with increasing anomaly depth (or increasing distance from the observation point). Note that this benchmark has uniform density so that the projection of the density from the nodes onto the quadrature points is exact and it does not introduce any smoothing of data which might occur in the presence of density discontinuities (e.g. air-water, water-crust, moho, etc.) inside a cell. $I=1$ seems to be the best compromise between accuracy (gravity acceleration errors are less than 0.01~mGal for all shells) and compute time for this experiment.
 
+\subsubsection{Thick shell gravity benchmark}
+\label{sec:benchmark-thick-shell-gravity}
+
+\textit{This section was contributed by Cedric Thieulot.}
+
+This benchmark tests the accuracy of the gravity field and gravitational potential computed by the gravity postprocessor inside and outside an Earth-sized planet (without its core) of constant density. The domain is a spherical shell with inner radius $R\textsubscript{inner}=3840~\si{\km}$ and outer radius $R\textsubscript{outer}=6371~\si{\km}$. The density is constant in the domain and set to $\rho_0=3300~\si{\kg\per\cubic\metre}$.
+
+First, let us calculate the exact profile which we expect the benchmark to reproduce. The gravitational potential $U$ of a spherically symmetric object satisfies the Poisson equation $\Delta U = 4\pi G \rho(\mathbf r)$. For a constant density shell, this equation can be solved analytically for the gravitational acceleration and potential inside and outside the planet.
+Outside the spherical shell the Poisson equation simplifies to the Laplace equation $\Delta U=0$:
+\begin{equation}
+\frac{1}{r^2} \frac{\partial }{\partial r} \left(r^2 \frac{\partial U}{\partial r} \right) = 0 
+\end{equation}
+The solution to this expression is:
+\begin{equation}
+g=\frac{\partial U}{\partial r} = \frac{C}{r^2} \label{eq:app1},
+\end{equation}
+where $C$ is a constant of integration.
+In order to avoid an infinite gravity field at $r=0$ (where the density is also zero in this particular setup of a shell), we need to impose $C=0$, i.e. the gravity is zero for $r\leq R\textsubscript{inner}$. Another way of arriving at the same conclusion is to realize that $g$ is zero at the centre of the body because the material around it exerts an equal force in every direction.
```

Comment:
I don't understand this argument. You derived this equation for the *outside* of the sphere, but here you talk about what happens at r=0?

Reply:
Oh, do you mean that this solution holds both *inside* and *outside* the sphere? If so, maybe it would be clearer to say that, and say that you have $C/r^2$ on the outside and $D/r^2$ on the inside, because these constants can be different. And that you want $D=0$ on the inside to keep the potential finite.

Reply:
yes, inside the shell rho=0 so Eq.(app1) holds (as it does outside). I added "Inside and outside the spherical shell (i.e. where $\rho=0$) the Poisson equation simplifies to the Laplace equation $\Delta U=0$ " a few lines above to clarify.

Reply:
Maybe "interior/exterior of the shell" would be a term that is clear?

Reply:
added: Inside ($r<R\textsubscript{inner}$) and outside ($r>R\textsubscript{outer}$) the spherical shell